categories/cookbook/13classes-objects-and-ties/13-04managing-class-data.p6

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env perl6
+
+use v6;
+
+=begin pod
+
+=TITLE Managing Instance data
+
+=AUTHOR uzluisf
+
+=head2 Problem
+
+You need a method invoked on behalf of the whole class, not just a single
+instance of the class.
+
+=head2 Solution
+
+In order to do this, a variable that is shared by all the instances of the class
+must be created. A variable of this kind is known as a I«class variable» and it
+uses the same syntax any other attribute, but it's declared as C«my» or C«our»,
+depending on the scope.
+
```

Comment:
Is there a possibility to quickly (under 1 or 2 sentences) extend the last sentence?

Reply:
That paragraph is pretty much a rewording of the documentation on class variable in which the last sentence is *"the second type of variables will be used across the class hierarchy."* I don't understand what that sentence is referring to yet so I didn't dare to include it. For this reason [I asked about it on #perl6](https://colabti.org/irclogger/irclogger_log/perl6?date=2019-06-30#l6) but it seems nobody was around at that time.

Reply:
I am not sure what "class hierarchy" is meant in the documentation, but to my understanding, it works this way:
```
class A { our $a = 15; my $b = 10 }
# "static" $a and $b
$A::a; # 15, because `our`
$A::b; # Any, because `my`
```
So
1)Would be nice to create a ticket in the documentation, asking to clarify this particular line, in fact, it is not very clear what is mean by "the second type", what's the first?
2)Maybe we can change "depending on the scope" to something like "depending on whether it will be exposed as a class package symbol or not"? Or something more clearer, to someone who is not too familiar with Perl 6, as I feel that my definition is technically correct, but may look like a black magic to someone. Any ideas?

Reply:
Thanks. I've created an issue in the docs. https://github.com/perl6/doc/issues/2876